tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -235,7 +235,7 @@ public function removeProductsFromOrder(string $productReference, string $orderR
         $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
         /** @var OrderProductForViewing[] $products */
         $products = $orderForViewing->getProducts()->getProducts();
-        foreach ($products as $product) {
+        foreach (array_reverse($products) as $product) {
```

Comment:
Since the default order is no `ASC` you don't need to reverse the array If you want to be sure you can force the parameter in GetOrderForViewing though
I think this might be related with the errors in the behat tests of specific prices